diff --git a/node_modules/code-push/script/acquisition-sdk.js b/node_modules/code-push/script/acquisition-sdk.js
index 5c1c7a5..6f023b4 100644
--- a/node_modules/code-push/script/acquisition-sdk.js
+++ b/node_modules/code-push/script/acquisition-sdk.js
@@ -53,7 +53,12 @@ var AcquisitionManager = /** @class */ (function () {
                 return;
             }
             try {
-                var responseObject = JSON.parse(response.body);
+                var responseObject;
+                try {
+                    responseObject = JSON.parse(response.body);
+                } catch {
+                    responseObject = response.body;
+                }
                 var updateInfo = responseObject.update_info;
             }
             catch (error) {
